src/tools/kdb/gen.cpp

```diff
@@ -65,11 +66,30 @@ int GenCommand::execute (Cmdline const & cl)
 		return 0;
 	}
 
+	/**
+	 * The parentKeyForTokenCalculation will be used to find the spec keys within the ks.
+	 * Because the token does not take into account metadata from other namespaces, it is only concerned with the spec: namespace.
+	 * Changes to the specification that was done in other namespaces is not the goal of the token mechanism.
+	 *
+	 * To really only take into account spec keys, the namespace of parentKeyForTokenCalculation is important!
+	 * There are 2 cases:
+	 *
+	 * 1. Keys loaded via plugin from file: Here, all keys within the ks will be in the cascading namespace.
+	 * Also, the ks will only contain keys from that file.
+	 * Thus, the parentKeyForTokenCalculation must be in cascading namespace as well.
+	 *
+	 * 2. Keys loaded from KDB: Here, the ks will contain keys from all namespaces (user:, spec: , ...)
+	 * For token calculation, only keys from the spec namespace are relevant.
+	 * Thus, the parentKeyForTokenCalculation must be in spec namespace.
+	 */
+	Key parentKeyForTokenCalculation (parentKeyName, KEY_END);
```

Comment:
I think you can avoid all of this `parentKeyForTokenCalculation` stuff, by simply requiring that `parentKeyName` starts with `spec:/`. We already check in `highlevel.cpp` that it starts with `spec:/` or `/` (and then generate the other variant). Theoretically other templates might not have this requirement, but for now `kdb gen` only makes sense for a `spec:/` parent, so we might as well enforce this properly.

Reply:
Thanks for your suggestion! Unfortunately, I don't have time to implement and test this change. 

I propose to merge this PR and make any further changes in a new one!

Reply:
Yeah, merging as is is totally fine for me.